### ISSUE TITLE
[UI] Implement "Updating in progress" indicator

### ISSUE THREAD

**Issue by Nahrain1** [amosproj/amos2024ss07-updating-flash-boot-loader#7]
## User story
1. As a user
2. I want to see the updating progress
3. So that I can improvise when the update is completed

## Acceptance criteria
* The updating progress indicator has to be simple and user-friendly
* The indicator has to fit possible display sizes
* Error handling mechanisms are there to catch issues that might occur during update and give user adequate feedback

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
